### PR TITLE
feat: switch to setuptools-scm for versioning

### PR DESCRIPTION
**Pull request by kemingy**
Signed-off-by: Keming <kemingy94@gmail.com>

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -85,35 +71,24 @@ def build_extension(self, ext: Extension):
...
-    classifiers=[
-        "Programming Language :: Python :: 3 :: Only",
-        "Programming Language :: Python :: 3.7",
-        "Programming Language :: Python :: 3.8",
-        "Programming Language :: Python :: 3.9",
-        "Programming Language :: Python :: 3.10",
-        "Programming Language :: Python :: Implementation :: CPython",
-        "Topic :: Software Development :: Libraries :: Python Modules",
-    ],
     python_requires=">=3.6",
```
Comment (lkevinzc) [inline setup.py:84]: 3.7?

Comment (kemingy) [reply setup.py:84]: We provide the PyPI package starting from `py3.7`. `py3.6` should also work but we don't guarantee it. If we set it to `py3.7`, I guess `py3.6` users can not install from the source distribution(not sure)?

**Comment by kemingy** [comment]
`mypy` is annoying, let's switch to `pyright`

**Comment by kemingy** [comment]
> `mypy` is annoying, let's switch to `pyright`

Will do this in another PR.

**Review by lkevinzc — APPROVED** [review]

**Pull request merged**